Added local deployment and initial startup guides for configdb and ob…

jnation3406 opened the pull request:
…s portal

This discusses how to run the docker-compose files, and how to do initial setup of those Apps in the admin interface. The cloud deployment section is not filled out but I thought we would put stuff on kubernetes/helm deployment in there. I also haven't added any specify "topics of interest* or tips and tricks for the deployment yet...

Commit:
Added local deployment and initial startup guides for configdb and obs portal

eheinrich commented on deployment/cloud_deployment.md:
```diff
@@ -0,0 +1,8 @@
+
+# Deployment
+
+This page will contain information on options for deploying the OCS in the cloud, such as on Amazon's AWS or Microsoft Azure.
```
I think changing "Amazon's AWS" -> "Amazon Web Services (AWS)" would sound a bit better, just cause the A already stands for Amazon. But I bet lots of people already say it the way you have it written, so I can be convinced otherwise.

eheinrich commented on deployment/configdb_setup.md:
```diff
@@ -0,0 +1,38 @@
+
+# Configuration Database Initial Setup
+
+When you start up a Configuration Database for the first time, you will need to login to it's /admin interface and fill in the initial representation of your observatory. Below we will walk through creating a single site, enclosure, and telescope with a single instrument on it, and the order in which you should create this. From there, you should know how to add additional telescopes or instrumen ...
```
"login" => "log in" when used as a verb https://grammarist.com/spelling/log-in-login/

"it's" => "its" when used to show possession

I think it's better to say admin interface without the slash, and then at the end say how to access the interface at the`/admin/` endpoint.

The above comments also apply to the first paragraph in the Observation Portal Set Up page

"... single instrument on it, and the order in which you should create this" => "... single instrument on it, showing the order in which you should create these objects"

eheinrich commented on deployment/configdb_setup.md:
```diff
@@ -0,0 +1,38 @@
+
+When you start up a Configuration Database for the first time, you will need to login to it's /admin interface and fill in the initial representation of your observatory. Below we will walk through creating a single site, enclosure, and telescope with a single instrument on it, and the order in which you should create this. From there, you should know how to add additional telescopes or instrumen ...
+
+1. Add a Site: Bolded fields cannot be left blank, but the most important fields to have correct are the 3-character **code** (usually the nearest airports 3-character code), and the **latitude** and **longitude** in decimal degrees. The **elevation**, **Timezone/Tz** and **restart** fields are not used by other OCS applications, so they are just their if you need them in your setup.
```
I think we should say that there aren't any pre-determined codes they need to choose from for their site code, just that they just need to come up with a unique 3 letter code for each of their sites. Reading that the site code must be correct makes me think that maybe I have to choose something specific. Something like:

"Bolded fields cannot be left blank, but the most important fields to fill in are the 3-character **code**, and the **latitude** and **longitude** in decimal degrees. The **code** can be any 3 letters, and each site should have a unique code. It might be convenient to use the 3-character code of the airport nearest the site, as long as two sites would not have the same code. The **elevation**, **Timezone/Tz** and **restart** fields are not used by other OCS applications, so they are just there if you need them in your setup."

"their" => "there"

eheinrich commented on deployment/configdb_setup.md:
```diff
@@ -0,0 +1,38 @@
+
+1. Add a Site: Bolded fields cannot be left blank, but the most important fields to have correct are the 3-character **code** (usually the nearest airports 3-character code), and the **latitude** and **longitude** in decimal degrees. The **elevation**, **Timezone/Tz** and **restart** fields are not used by other OCS applications, so they are just their if you need them in your setup.
+
+2. Add an Enclosure: You will just need a 4-character enclosure **code** (could be dome or doma/domb, or anything else that represents the type of enclosure), and then link it to the site you created in step 1.
```
Suggest: "Fill in a 4-character enclosure **code**, and then link it to the site you created in step 1. The 4-character enclosure code can be anything of your choice that describes the type of enclosure. For example if you have two enclosures of the same type at the same site, and you are calling the type `dom`, you could use enclosure codes `doma` and `domb`."

eheinrich commented on deployment/configdb_setup.md:
```diff
@@ -0,0 +1,38 @@
+
+2. Add an Enclosure: You will just need a 4-character enclosure **code** (could be dome or doma/domb, or anything else that represents the type of enclosure), and then link it to the site you created in step 1.
+
+3. Add a Telescope: The 4-character **code** should be something descriptive about the telescope, such as it's aperture (i.e. 1m4a/1m4b for two 1.4m telescopes). There are several important properties here that are used automatically by the rest of the OCS. The **latitude** and **longitude**, **horizon** limit, positive and negative **hour angle limits**, and radial **zenith blind spot** fields a ...
```
"it's" => "its"

"computing request durations" => "computing observation request durations"

eheinrich commented on deployment/configdb_setup.md:
```diff
@@ -0,0 +1,38 @@
+
+5. Add Optical Elements: You will need to add an optical element for each specific filter, slit, grating, diffuser, etc. that is an option for your instruments. We will group them into categories in the next step, but for now just add each specific one, with a **code** that users will use to specify that element, and a human readable **name** that will be displayed in the user interface.
+
+6. Add Optical Element Groups: For each category of **Optical Elements** that you created above, such as filters, slits, etc., you will make one Optical Element Group representing that category. The **type** field contains the pluralized category, like *filters* or *slits*, while the **name** can be used to give some identifying information to this group. Then select one or more **optical element ...
```
"computing request durations" => "computing observation request durations"

eheinrich commented on deployment/configdb_setup.md:
```diff
@@ -0,0 +1,38 @@
+
+6. Add Optical Element Groups: For each category of **Optical Elements** that you created above, such as filters, slits, etc., you will make one Optical Element Group representing that category. The **type** field contains the pluralized category, like *filters* or *slits*, while the **name** can be used to give some identifying information to this group. Then select one or more **optical element ...
+
+7. Add a Camera: You must select the **Camera Type** you created above for this camera, as well as give it a unique **code** and default **orientation** of the camera ccd. You will also select zero or more **optical element groups** that apply to this camera. These will be used for automatic validation of observing requests on this camera.
```
"ccd" => "CCD"

eheinrich commented on deployment/configdb_setup.md:
```diff
@@ -0,0 +1,38 @@
+
+7. Add a Camera: You must select the **Camera Type** you created above for this camera, as well as give it a unique **code** and default **orientation** of the camera ccd. You will also select zero or more **optical element groups** that apply to this camera. These will be used for automatic validation of observing requests on this camera.
+
+8. Add an Instrument Category: The category will be reported by the Observation Portal in it's description of the available instruments, and can be used by the frontend. Las Cumbres Observatory uses two categories, *IMAGE* and *SPECTRA*, which broadly represent imager and spectrograph type cameras. Our frontend uses this information to provide specific behaviour based on if an instrument is a spe ...
```
"it's" => "its"

eheinrich commented on deployment/configdb_setup.md:
```diff
@@ -0,0 +1,38 @@
+
+9. Add an Instrument Type: Each type of instrument must have a corresponding **Instrument Type** created to specify its instrument level overheads and default **acceptability threshold** for that instrument.
+
+10. Add some Mode Types: These are used for creating **Generic Mode Groups**. The Observation Portal expects to validate modes of these specific types, so at a minimum these should be added: **readout**, **guiding**, **acquisition**, **rotator**, and **exposure**. You can create other mode types if you have other modes that you will add custom validation to your Observation Portal for.
```
Should everyone need to set exposure mode? We only use it for muscat and it's set in the extra params, so doesn't feel like it's part of the base. I also would say that probably rotator modes are not required either. If they don't have any instruments that have rotators, they won't need it.

eheinrich commented on deployment/configdb_setup.md:
```diff
@@ -0,0 +1,38 @@
+
+12. Add Generic Mode Groups: Once you have some **Generic Modes** of a certain mode **type**, you can create a group to apply those modes to a specific **Instrument Type**. One or more modes can be in the group, and an optional default can be specified to be applied if the user does not specify a mode of that type.
+
+13. Add an Instrument: You must specify a **code**, which will be used to specify the instrument an observation is scheduled to take data on. The only automatic **state** used is *SCHEDULABLE*, which means the instrument is available for scheduling. You must specify which **telescope** the instrument is on, and give it one or more **science cameras** that make up the instrument, as well as one ** ...
```
Can add that it there is no autoguider camera, it can be set to the science camera

eheinrich commented on deployment/configdb_setup.md:
```diff
@@ -0,0 +1,38 @@
+
+13. Add an Instrument: You must specify a **code**, which will be used to specify the instrument an observation is scheduled to take data on. The only automatic **state** used is *SCHEDULABLE*, which means the instrument is available for scheduling. You must specify which **telescope** the instrument is on, and give it one or more **science cameras** that make up the instrument, as well as one ** ...
+
+This is a lengthy process for the initial setup, but once that is done you should only need to add more instruments or modes or tweak overhead times, which should take considerable less time.
```
"considerable" => "considerably"

eheinrich commented on deployment/configdb_setup.md:
```diff
@@ -0,0 +1,38 @@
+
+This is a lengthy process for the initial setup, but once that is done you should only need to add more instruments or modes or tweak overhead times, which should take considerable less time.
+
+See the [Observation Portal Setup]({% link deployment/obs_portal_setup.md %}) section for information on how to setup a new Observation Portal to begin accepting Observation Requests.
```
"setup" => "set up" http://www.enhancemywriting.com/setup-vs-set-up

eheinrich commented on deployment/index.md:
```diff
@@ -0,0 +1,8 @@
+
+# Deployment
+
+These sections will discuss options for a production deployment of the OCS applications, as well as information on intial setup for the main OCS applications. To begin, go to [Local Deployment]({% link deployment/local_deployment.md %}) for instructions on how to run the OCS applications on a single machine or network of machines, or go to [Cloud Deployment]({% link deployment/cloud_deployment.md ...
```
"... provider such as AWS in kubernetes with helm charts." => "... provider such as AWS." I'm sure we will provide some helm charts, but I guess technically they could also just buy an instance in AWS or something and deploy their apps on there just like they would a normal machine.

Add on a paragraph about initializing the different applications, something like:

"After deploying your applications, you can find information on how to initialize the databases of the different applications in the set up sections."

jnation3406 replied:
I had the kubernetes w/helm charts statement because I figured that is what we would provide for the "cloud" section. I guess maybe it's more correct to just call it a kubernetes deployment section? If its just using normal machines in AWS to run the docker-compose files then they can just use the local deployment section.

jnation3406 replied:
I've just renamed the cloud deployment to be kubernetes deployment since I think that fits better with what we want to describe there.

eheinrich replied:
Sounds good 👍

eheinrich commented on getting_started.md:
```diff
@@ -9,6 +9,8 @@ The Observatory Control System (OCS) provides an interface that allows astronome
 
 There are many different ways to run the projects that are part of the OCS. The simplest way to get up and running and to start evaluating it for your own purposes is the run the [OCS example project](https://github.com/observatorycontrolsystem/ocs_example).
 
+{% include notification.html message="The OCS example project is meant for evaluation purpose only and should not be used in a production setup. It inserts test data into the database, and the database is cleared and reset each time the example is taken down and restarted. Please check out the deployment section to learn how to setup a production environment." %}
```
"evaluation purpose only" => "evaluation purposes only"

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,71 @@
+
+# Local or Network Deployment
+
+This page will give some instructions on how to run the OCS applications in a production environment on one or more servers connected through a local network. If you are interested in a persistant, high availability cloud deployment procedure, check out the [Cloud Deployment]({% link deployment/cloud_deployment.md %}) section.
```
"persistant" => "persistent"

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,71 @@
+
+## Database Setup
+
+Most of the OCS application depend on a postgresQL database (with the postGIS plugin installed for the Science Archive). You can share the same postgres instance with all the applications, they will just each need their own database created, but they could share credentials for simplicity. Your options for running postgres are:
```
"Most of the OCS application" => "Most of the OCS applications"

"postgresQL" => "postgreSQL"

"postgres" => "postgreSQL"

Suggest: "database (with the postGIS plugin installed for the Science Archive)" => "database. The Science Archive additionally requires the postGIS plugin be installed in its postgreSQL database."

Suggest: "You can share the same postgres instance with all the applications, they will just each need their own database created, but they could share credentials for simplicity" => "It is possible to use the same postgreSQL instance across all the applications for a simpler setup, they will just each need their own database created."

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,71 @@
+
+Most of the OCS application depend on a postgresQL database (with the postGIS plugin installed for the Science Archive). You can share the same postgres instance with all the applications, they will just each need their own database created, but they could share credentials for simplicity. Your options for running postgres are:
+
+* Use an existing postgres instance on your local network
```
"postgres" => "postgreSQL"

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,71 @@
+Most of the OCS application depend on a postgresQL database (with the postGIS plugin installed for the Science Archive). You can share the same postgres instance with all the applications, they will just each need their own database created, but they could share credentials for simplicity. Your options for running postgres are:
+
+* Use an existing postgres instance on your local network
+* Install postgres locally on your machine
```
"postgres" => "postgreSQL"

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,71 @@
+
+* Use an existing postgres instance on your local network
+* Install postgres locally on your machine
+* Run a postgres/postgis docker container
```
Suggest: "Run a postgreSQL docker container with postGIS installed."

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,71 @@
+* Install postgres locally on your machine
+* Run a postgres/postgis docker container
+
+To start a postgresQL instance with postgis on a machine with docker forwarding it's port 5432 for local usage, you can use:
```
"postgresQL" => "postgreSQL"

"postgis" => "postGIS"

"it's" => "its"

eheinrich commented on deployment/local_deployment.md:
````diff
@@ -0,0 +1,71 @@
+docker run --name ocs-postgis -p 0.0.0.0:5432:5432 -e POSTGRES_PASSWORD=postgres -d postgis/postgis
+```
+
+{% include notification.html message="By default, postgresQL within a docker container will store it's data in the docker containers filesystem. If you want to persist the database, should should either create a docker volume or volume mount the postgres data directory in the container to a folder on your local filesystem when you start the container." %}
````
"postgresQL" => "postgreSQL"

"it's" => "its"

"in the docker containers filesystem" => "in the docker container's filesystem"

"postgres" => "postgreSQL"

I think it'd be good to link to some docker docs here https://docs.docker.com/storage/volumes/

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,71 @@
+
+{% include notification.html message="By default, postgresQL within a docker container will store it's data in the docker containers filesystem. If you want to persist the database, should should either create a docker volume or volume mount the postgres data directory in the container to a folder on your local filesystem when you start the container." %}
+
+After you have a postgres instance running that you can connect to with a username/password (Assuming postgres/postgres here, change as needed), create a db for each OCS application you will use:
```
"postgres" => "postgreSQL"

Suggest: "with a username/password (Assuming postgres/postgres here, change as needed), create a db for each OCS application you will use:" => "with a username and password, create a database for each OCS application you will use. These instructions assume the username and password are both set to the value `postgres`, please update as needed:"

eheinrich commented on deployment/local_deployment.md:
````diff
@@ -0,0 +1,71 @@
+After you have a postgres instance running that you can connect to with a username/password (Assuming postgres/postgres here, change as needed), create a db for each OCS application you will use:
+
+```bash
+createdb --host 0.0.0.0 -p5432 -Upostgres -W my_db_name
````
Why did you choose to use `0.0.0.0` for the example instead of `localhost` or `127.0.0.1`?

jnation3406 replied:
no real reason, I think they work interchangeably. Do you have a preference?

eheinrich replied:
My preference would be when starting up a web server have it listen on `0.0.0.0` so that it handles traffic sent to any of the addresses that the host has defined for itself, but when setting a url to point to a service or the db or whatever, use a single address like `127.0.0.1`. I came to that conclusion after reading [this page](https://www.howtogeek.com/225487/what-is-the-difference-between-127.0.0.1-and-0.0.0.0/#:~:text=0.0%3F-,127.0.,particular%20address'%20place%20holder) which defines `0.0.0.0` as "a non-routable meta-address used to designate an invalid, unknown, or non-applicable target (a ‘no particular address’ place holder)" and requests to it will be sent to the default route... not sure exactly how the default route is defined, but it seems safer to explicitly say the db is located at and should be accessed using `127.0.0.1` instead of `0.0.0.0` in case the default route resolves to something else.

eheinrich commented on deployment/local_deployment.md:
````diff
@@ -0,0 +1,71 @@
+To start a postgresQL instance with postgis on a machine with docker forwarding it's port 5432 for local usage, you can use:
+
+```bash
+docker run --name ocs-postgis -p 0.0.0.0:5432:5432 -e POSTGRES_PASSWORD=postgres -d postgis/postgis
````
Why did you choose to use `0.0.0.0` for the example?

The port is specified twice.

jnation3406 replied:
It's a port mapping from localhost:5432 to container 5432 so I think that is correct...

eheinrich replied:
Ah gotcha, I haven't seen that format for the `-p` argument before, I thought it was just port:port.

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,71 @@
+
+## Docker-Compose Templates
+
+The preferred way to run each OCS application is using Docker. This way, you do not need to worry about installing the proper dependencies, or debugging strange interactions on your specific system. Template docker-compose.yaml files are provided within each OCS application repo to help getting setup to run the application in production. The docker-compose files should be setup in the repo to wor ...
```
Suggestion for this section:


The preferred way to run each OCS application is using Docker. This way, you do not need to worry about installing the proper dependencies, or debugging strange interactions on your specific system. Each OCS application provides a Docker image that is available on [Docker Hub](https://hub.docker.com/u/observatorycontrolsystem), or you can build the Docker images yourself using the `Dockerfile`s in the OCS application repositories.

One way to configure and run one or more Docker containers is by using Docker Compose. A set of services that are run via Docker Compose are specified using a Docker Compose file. A single Docker Compose file can technically be used to start up all the different applications of the OCS, but we would not recommend doing that for production deploys. One issue with starting up all components together is that if any component needs to be upgraded, then all of the components would have to be restarted. Another issue is that that all applications would need to run on a single machine. Instead we recommend logically splitting up all the applications into separate Docker Compose files. Template `docker-compose.yml` files are provided within each OCS application repository to help you get started.

The Docker Compose files in each repository are set up to work and connect with one another given a purely local deploy of all applications. If you are installing the OCS applications on different networked servers, you will need to update their environment variables to point to the correct location for each of their connections.

(Information about updating the environment variables to point at the preferred database etc goes here)

When starting the different Docker Compose stacks, note that the Observation Portal requires a running ConfigDB to start up, the Adaptive Scheduler requires both the Observation Portal and ConfigDB to be running, and the Science Archive requires the Observation Portal to be running if you are using the user accounts in the Observation Portal for user authentication in the Science Archive. The Downtime Database can be started on its own.

If you are new to Docker and Docker Compose, or if you would like to know more, check out the following resources:
- [Information on how to install Docker](https://docs.docker.com/get-docker/)
- [Highly recommended tutorial to get you started using Docker and get you familiar with it](https://docs.docker.com/get-started/)
- [Brief overview of Docker Compose](https://docs.docker.com/compose/)

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,71 @@
+
+## Application Initial Setup
+
+The OCS applications are modular, so you can pick and choose which pieces you need in your setup. At a very minimum though, you will likely want to run the Observation Portal for its User accounts and authentication, and the Configuration Database to define the properties of your telescopes and instruments. The Configuration Database depends on the Observation Portal for account authentication, s ...
```
"django" => "Django"

Suggest: "This can be done by running the Observation Portal's custom `create_user` Django management command from within the Observation Portal container, updating the username and password as needed:"

eheinrich replied:
There are some of comments that I left (like this one) that weren't resolved or commented on... did you see these?

eheinrich commented on deployment/local_deployment.md:
````diff
@@ -0,0 +1,71 @@
+OAUTH_CLIENT_SECRET=configdb_application_client_secret
+```
+
+{% include notification.html message="Note if you are running other applications that connect to the Observation Portal for authentication, such as the Science Archive or Downtime Database, you will need to run the create_application management command once for each of those applications as well to set up their credentials"
````
Include that other oauth apps need to have different client IDs and client secrets

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,71 @@
+status="is-info"
+icon="fas fa-exclamation-triangle" %}
+
+You should then be ready to start the Configuration Database container and have access to it's admin interface through your Observation Portal account credentials. See the [ConfigDB Setup]({% link deployment/configdb_setup.md %}) section for information on how and in what order to create your telescope representation in the Configuration Database.
```
"it's" => "its"

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,71 @@
+* Use an existing postgres instance on your local network
+* Install postgres locally on your machine
+* Run a postgres/postgis docker container
+
```
Suggest to add a paragraph here suggesting database backups, like: "When running in production, it is recommended that database backups are created periodically. How frequently you create backups will depend on your needs. For example, creating one backup every day, and saving the last seven backups that were created would ensure that you would lose at most one day's worth of data, and that you could revert to any snapshot of the database taken within the last seven days. For information on how to create database backups, you can refer to the postgreSQL documentation [here](https://www.postgresql.org/docs/9.1/backup.html)."

eheinrich commented on deployment/local_deployment.md:
````diff
@@ -0,0 +1,71 @@
+createdb --host 0.0.0.0 -p5432 -Upostgres -W my_db_name
+```
+
+The docker-compose templates within each OCS application should then be modified to set the proper environment variables to point to your database. Note you should change the DB name, user/password, and host/port to the proper values for your setup:
````
I think this should go in the Docker-Compose Templates section because at this point we haven't yet introduced that each repo has a docker compose template and this section is just DB setup

eheinrich commented on deployment/obs_portal_setup.md:
```diff
@@ -0,0 +1,21 @@
+
+# Observation Portal Initial Setup
+
+There are a things that must be added to a newly deployed Observation Portal to begin accepting Observation Requests. You should already have a superuser account created which you can use to login to the /admin interface of the Observation Portal. All steps are accomplished by clicking on the corresponding model in the /admin interface and clicking the "Add a *" button in the upper right corner.
```
Mention that a prerequisite is having configdb set up already, and refer to that section if they have not already done so.

eheinrich commented on deployment/obs_portal_setup.md:
```diff
@@ -0,0 +1,21 @@
+
+## Setup a Proposal
+
+The first step to accepting Requests is to setup a proposal with time on it for the **Instrument Types** you created in the Configuration Database.
```
"Requests" => "Observation Requests"

"setup" => "set up"

"proposal with time on it for the" => "proposal with time allocation to it for the"

eheinrich commented on deployment/obs_portal_setup.md:
```diff
@@ -0,0 +1,21 @@
+
+The first step to accepting Requests is to setup a proposal with time on it for the **Instrument Types** you created in the Configuration Database.
+
+1. Add a Semester: This represents an Observing Semester at an Observatory, which is a period of time during which time on instruments is allocated and available for use for proposals. These should be non-overlapping, but you can create as many of these as you want, i.e. yearly, bi-yearly, or just have a single one with a time range of all time.
```
"for use for proposals" => "for use by proposals"

eheinrich commented on deployment/obs_portal_setup.md:
```diff
@@ -0,0 +1,21 @@
+
+1. Add a Semester: This represents an Observing Semester at an Observatory, which is a period of time during which time on instruments is allocated and available for use for proposals. These should be non-overlapping, but you can create as many of these as you want, i.e. yearly, bi-yearly, or just have a single one with a time range of all time.
+
+2. Add a Proposal: This can have time allocated to it on instrument types during observing semesters, which is available for use by any Principle Investigator (PI) or Co-Investigator (COI) on the proposal. At a minimum, you must specify the **Id** of the proposal, which will be used when submitting Requests on that proposal, and then the **tac priority**, which is used by the scheduler when evalu ...
```
Suggest: "This can have time allocated to it on instrument types during observing semesters, which is available for use by any Principle Investigator (PI) or Co-Investigator (COI) on the proposal" => "Members of a proposal, which include the proposal's Principle Investigator (PI) or Co-Investigator(s) (CoI), are allowed to submit Observation Requests for instruments types that their proposal has been allocated time on for a particular observing semester."

Suggest: "which is used by the scheduler when evaluating where to place Observations" => "which is used by the Adaptive Scheduler when it is calculating the observing schedule."

Suggest: "The **direct submission** field should be checked if you want to be able to submit Observations directly with this Proposal through the Observation Portal /api/schedule/ API - otherwise the Proposal will only be allowed to submit Requests" => "The **direct submission** field should be checked if you want to allow members of the proposal to to create Observations directly using the Observation Portal `/api/schedule/` API - otherwise members of the Proposal will only be allowed to submit Observation Requests."

"then select the Obsering **Semester** this will apply to" => "then select the Observing **Semester** this Time Allocation will apply to"

eheinrich commented on deployment/obs_portal_setup.md:
```diff
@@ -0,0 +1,21 @@
+
+2. Add a Proposal: This can have time allocated to it on instrument types during observing semesters, which is available for use by any Principle Investigator (PI) or Co-Investigator (COI) on the proposal. At a minimum, you must specify the **Id** of the proposal, which will be used when submitting Requests on that proposal, and then the **tac priority**, which is used by the scheduler when evalu ...
+
+3. Add a Proposal Membership: Each Proposal should have at least one PI, who will be able to control the time allocation and access of COIs on the proposal under them. To add this initial PI, select the **proposal** and **user** by Id using the search button, and then select a **role** of Principle Investigator. The **time limit** field is used to limit the specific user to a specific amount of t ...
```
Suggest: "Each Proposal should have at least one PI, who will be able to control the time allocation and access of COIs on the proposal under them" => "Each Proposal should have at least one PI, who will be able manage the proposal's COIs, including adding and removing COIs and setting limits on how much of the proposal's allocated observing time they can use."

"Id" => "ID"

"... limit the specific user to a specific amount of time on that ..." => ".. limit the specific user to a specific amount of time they are allowed to use on that ..."

eheinrich commented on deployment/obs_portal_setup.md:
```diff
@@ -0,0 +1,21 @@
+
+3. Add a Proposal Membership: Each Proposal should have at least one PI, who will be able to control the time allocation and access of COIs on the proposal under them. To add this initial PI, select the **proposal** and **user** by Id using the search button, and then select a **role** of Principle Investigator. The **time limit** field is used to limit the specific user to a specific amount of t ...
+
+All users that are members should now by able to submit Observing Requests specifying this newly created proposal. New user accounts can be requested by users hitting the site, and the superuser/administrator can approve those accounts. Those accounts will then need to be added onto a proposal to be able to submit requests and see the proposals data.
```
"New user accounts can be requested by users hitting the site ..." => "New user accounts can be requested by users by using the Observation Portal's registration form ..."

Actually, regarding creating new users, I have a question about this part of the sentence: "and the superuser/administrator can approve those accounts". Are you talking about users registring new accounts or something else?

"submit requests" => "submit Observation Requests"

Also suggest to add sentence "A PI can also invite new members to join their proposal even if that user does not yet have an account in the Observation Portal - that user will receive an email letting they know that they have been invited to join, and that they can register an account."

jnation3406 replied:
I'm talking about registering accounts... I assumed someone submits there details to register and then Nikolaus has to accept their account registration for their account to be active? If it doesn't work like that I can remove that part...

eheinrich replied:
I think it's possible to require an admin to approve an account, but currently the flow is user submits registration form -> user gets email letting them know they submitted a request to register with a link to activate their account -> user clicks activation link and then they have an account

eheinrich reviewed (commented):
Nice! I think pretty much everything that I had wanted to add is in here one form or another. I did leave a lot of comments, so I'll wait to approve until after you take a look at them in case there's any issue, and also so I can read through these docs one more time after revisions have gone in.

Commit:
Updated from review comments

jnation3406 commented:
> Nice! I think pretty much everything that I had wanted to add is in here one form or another. I did leave a lot of comments, so I'll wait to approve until after you take a look at them in case there's any issue, and also so I can read through these docs one more time after revisions have gone in.

Okay, I think I've made changes addressing all your comments. Take a look and let me know if there is anything else.

Commit:
Last few review comments

eheinrich commented on deployment/local_deployment.md:
````diff
@@ -0,0 +1,88 @@
+docker run --name ocs-postgis -p 0.0.0.0:5432:5432 -e POSTGRES_PASSWORD=postgres -d postgis/postgis
+```
+
+{% include notification.html message="By default, postgreSQL within a docker container will store its data in the docker container's filesystem. If you want to persist the database, should should either create a docker volume or volume mount the postgreSQL data directory in the container to a folder on your local filesystem when you start the container. Check out the [docker docs](https://docs.do ...
````
"should should" => "you should"

eheinrich commented on deployment/local_deployment.md:
```diff
@@ -0,0 +1,88 @@
+
+## Docker-Compose Templates
+
+The preferred way to run each OCS application is using Docker. This way, you do not need to worry about installing the proper dependencies, or debugging strange interactions on your specific system. Template docker-compose.yaml files are provided within each OCS application repo to help getting setup to run the application in production. The docker-compose files should be setup in the repo to wor ...
```
Delete this paragraph (the one on line 38). The same information is covered in the subsequent paragraphs.

eheinrich reviewed (approved):
Looks good!

Commit:
fixed review comments
